mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
+
+  return (targetPort == "" || isDefaultPortForProtocol) ?
+          "" : ":" + targetPort;
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
+    parent.appendChild(el);
+
+  return el;
+}
+
+function createHelperIframe(name) {
+  return createElement("iframe", {"name": name}, document.body);
+}
+
+function queryIframe(url) {
+  createElement("iframe", {"src": url}, document.body);
+}
+
+function queryImage(url) {
+  createElement("img", {"src": url}, document.body)
+}
+
+function queryXhr(url) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = "json";
+  xhr.send();
+}
+
+function queryWorker(url) {
+  var worker = new Worker(url);
+  console.log("worker", url);
+  worker.postMessage('');
+}
+
+function queryFetch(url) {
+  try {
+    fetch(url);
+  } catch (ex) {
+    console.error("Failed: fetch constructor.", ex);
```

Comment:
I believe I put this here so that the promises don't fail due to the exception thrown in the test.
Added a TODO to check it out.